### PR TITLE
android: fix all linter warnings and treat warnings as errors

### PR DESCRIPTION
**Pull request by barnstar**
#Updates tailscale/corp#22284

Fixes and/or explicitly suppresses all linter warnings and we will now fail the build if new warnings are introduced.

### CODE REVIEW

**Review by agottardo — APPROVED** [review]

File: android/src/main/java/com/tailscale/ipn/ui/theme/Theme.kt [hunk 1]
```diff
@@ -44,7 +44,7 @@ fun AppTheme(useDarkTheme: Boolean = isSystemInDarkTheme(), content: @Composable
           // margins in list items.
           bodyMedium = MaterialTheme.typography.bodyMedium.copy(fontSize = 16.sp))
 
-  val systemUiController = rememberSystemUiController()
+  @Suppress("deprecation") val systemUiController = rememberSystemUiController()
```
Comment (kari-ts) [inline android/src/main/java/com/tailscale/ipn/ui/theme/Theme.kt:47]: can you create an issue/todo to use the new API? looks like the recommendation is WindowInsetsControllerCompat or Activity.enableEdgeToEdge per https://google.github.io/accompanist/systemuicontroller/

File: android/src/main/java/com/tailscale/ipn/QuickToggleService.java [hunk 2]
```diff
@@ -9,6 +9,7 @@
 import android.service.quicksettings.Tile;
 import android.service.quicksettings.TileService;
 
+@SuppressWarnings("deprecation")
```
Comment (kari-ts) [inline android/src/main/java/com/tailscale/ipn/QuickToggleService.java:12]: can you move this to the line with the deprecated API and if relevant, add a migration todo?

File: android/src/main/java/com/tailscale/ipn/ui/util/AndroidTVUtil.kt [hunk 3]
```diff
@@ -15,7 +15,7 @@ import com.tailscale.ipn.ui.util.AndroidTVUtil.isAndroidTV
 object AndroidTVUtil {
   fun isAndroidTV(): Boolean {
     val pm = UninitializedApp.get().packageManager
-    return (pm.hasSystemFeature(PackageManager.FEATURE_TELEVISION) ||
+    return (pm.hasSystemFeature(@Suppress("deprecation") PackageManager.FEATURE_TELEVISION) ||
```
Comment (kari-ts) [inline android/src/main/java/com/tailscale/ipn/ui/util/AndroidTVUtil.kt:18]: add comment to migrate to FEATURE_LEANBACK pls

**Review by kari-ts — APPROVED** [review]
thanks! I'll close https://github.com/tailscale/tailscale-android/pull/519 as this is more complete

File: android/src/main/java/com/tailscale/ipn/ui/util/AndroidTVUtil.kt (hunk 3 above)
Comment (barnstar) [reply android/src/main/java/com/tailscale/ipn/ui/util/AndroidTVUtil.kt:18]: We check both...  It's just for compatibility with old versions.

**Pull request merged**